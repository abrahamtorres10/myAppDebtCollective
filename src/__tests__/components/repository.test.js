import React from "react"
import renderer from "react-test-renderer"
import Repository from "../../components/repository"

describe("Repository test suite", () => {
  it("renders correctly", () => {
    const baseProps = {
      name: "Test repository name",
      description: "Test repository description",
      forkCount: 3,
      createdAt: new Date(2019, 2, 2),
      url: "www.google.com",
    }
    const tree = renderer.create(<Repository {...baseProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders 'Description not available' if description is empty", () => {
    const baseProps = {
      name: "Test repository name",
      description: "",
      forkCount: 3,
      createdAt: new Date(2019, 2, 2),
      url: "www.google.com",
    }
    const tree = renderer.create(<Repository {...baseProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
