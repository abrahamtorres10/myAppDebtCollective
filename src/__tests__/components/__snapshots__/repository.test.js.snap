// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository test suite renders 'Description not available' if description is empty 1`] = `
<div
  className="card"
>
  <div
    className="content"
  >
    <h3>
      Test repository name
    </h3>
    <div
      className="subtitle"
    >
      03/02/2019
    </div>
    <p
      className="description"
    >
      Description not available
    </p>
    <div
      className="details"
    >
      <div
        className="inner"
      >
        <div>
          <a
            className="button"
            href="www.google.com"
            id="repoUrl"
          >
            Go !
          </a>
        </div>
        <div
          className="options"
        >
          <div>
            <a>
              <i
                class="fas fa-code-branch"
              />
               
              3
               forks
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Repository test suite renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="content"
  >
    <h3>
      Test repository name
    </h3>
    <div
      className="subtitle"
    >
      03/02/2019
    </div>
    <p
      className="description"
    >
      Test repository description
    </p>
    <div
      className="details"
    >
      <div
        className="inner"
      >
        <div>
          <a
            className="button"
            href="www.google.com"
            id="repoUrl"
          >
            Go !
          </a>
        </div>
        <div
          className="options"
        >
          <div>
            <a>
              <i
                class="fas fa-code-branch"
              />
               
              3
               forks
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
